### ISSUE TITLE
MySQL: Figure out how to handle tests that modify schema

### ISSUE THREAD

**Issue by killercup** [diesel-rs/diesel#661]
Our test suite for Postgres and SQLite assumes we can modify the db schema inside a transaction. This does not work on MySQL (it commits the transaction when you try to modify the schema).

We need to figure out a way to have those tests run on MySQL without introducing a bunch of overhead.

(To find affected tests, grep for `FIXME: Figure out how to handle tests that modify schema`)

**Comment by jraffeiner** [comment]
Since `PostgreSQL`, `Sqlite`, `Mysql` and `Mariadb` support CREATE TEMPORARY TABLE, and all shadow persistent tables with the same name, would that be a possible solution?

**Comment by weiznich** [comment]
Yes, that's the correct way forward. In fact we already use this pattern for some of the newer tests, but never came around adjusting the existing old tests. Any help there would be welcome.
